src/DatePicker/DatePicker.js

```diff
@@ -282,20 +398,36 @@ class DatePicker extends Component {
 
     const {prepareStyles} = this.context.muiTheme;
     const formatDate = formatDateProp || this.formatDate;
+    const rawDate = this.state.date instanceof Date ?
+      formatDate(this.state.date) :
+      this.state.date;
+    const inputError = rawDate !== undefined && !(this.state.date instanceof Date) ?
+      'Enter a valid date' :
+      this.props.errorText;
+    const hintText = keyboardEnabled && this.state.keyboardActivated ? 'yyyy-mm-dd' : this.props.hintText;
 
     return (
-      <div className={className} style={prepareStyles(Object.assign({}, style))}>
+      <div ref="root" className={className} style={prepareStyles(Object.assign({}, style))}>
         <TextField
           {...other}
-          onFocus={this.handleFocus}
+          onFocus={this.handleInputFocus}
+          onBlur={this.handleInputBlur}
+          onKeyDown={this.handleKeyDown}
```

Comment:
That's a breaking change, users can no longer override the `onKeyDown` property.
That's definitely not the only breaking change.